devops/jobs/ConfigurationWatcher.groovy

```diff
@@ -0,0 +1,154 @@
+            assert extraVars.containsKey('APPS_TO_CONFIG'): "Please define APPS_TO_CONFIG"
+
+            def config_branch = 'master'
+
+            // to space separated list as string
+            def apps = extraVars.get("APPS_TO_CONFIG").keySet().join(",").replaceAll(',', " ")
+
+            // inject APPS as an environment variable to be used by trigger-builds.sh
+            environmentVariables {
+                env('APPS', apps)
+            }
+
+            wrappers common_wrappers
+            // add credentials to log in to DockerHub; ID refers to credential ID as set up in Jenkins
+            wrappers {
+                credentialsBinding {
+                    file("CONFIG_JSON_FILE", extraVars.get("CONFIG_JSON_FILE_CREDENTIAL_ID", "docker-config-json"))
+                }
+            }
+            logRotator common_logrotator
+
+            def access_control = extraVars.get('ACCESS_CONTROL',[])
+            access_control.each { acl ->
+                common_read_permissions.each { perm ->
+                    authorization {
+                        permission(perm,acl)
+                    }
+                }
+            }
+
+            // for GitHub Webhooks
+            properties {
+                githubProjectUrl(extraVars.get("EDX_REPO_ROOT") +  "configuration/")
+            }
+
+            scm {
+                // check out edx/configuration repository from GitHub
+                git {
+                    remote {
+                        url(extraVars.get("CONFIGURATION_REPO_URL"))
+                        branch(config_branch)
+                    }
+                    extensions {
+                        relativeTargetDirectory('configuration')
+                    }
+                }
+                
+            }
+
+            triggers {
+                // due to a bug or misconfiguration, jobs with default branches with
+                // slashes are indiscriminately triggered by pushes to other branches.
+                // For more information, see:
+                // https://openedx.atlassian.net/browse/TE-1921
+                // for commits merging into master, trigger jobs via github pushes
+                    if (config_branch == 'master') {
+                        githubPush()
+                    }
+                // for all other jobs in this style, poll github for new commits on
+                // the 'defaultBranch' every 10 minutes
+                    else {
+                        scm('H/10 * * * *')
+                    }
+            }
+            
+            // run the trigger-builds shell script in a virtual environment called venv
+            steps {
+                virtualenv {
+                    name('venv')
+                    nature('shell')
+                    command dslFactory.readFileFromWorkspace('devops/resources/trigger-builds.sh')
+                }
+
+                 // inject environment variables defined in the temp_props file (TO_BUILD)
```

Comment:
Can you add more comments about where/how temp_props is generated?  I figured it out with grep, but would love a shortcut.